src/domain/auth-rebac/domain/src/repos/rebac_repository.rs

```diff
@@ -0,0 +1,494 @@
+
+    async fn get_relations_between_entities(
+        &self,
+        subject_entity: &Entity,
+        object_entity: &Entity,
+    ) -> Result<Vec<Relation>, GetRelationsBetweenEntitiesError>;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Errors
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Error, Debug)]
+pub enum SetEntityPropertyError {
+    #[error(transparent)]
+    Internal(InternalError),
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Error, Debug)]
+pub enum DeleteEntityPropertyError {
+    #[error(transparent)]
+    EntityNotFound(EntityNotFoundError),
+
+    #[error(transparent)]
+    PropertyNotFound(EntityPropertyNotFoundError),
+
+    #[error(transparent)]
+    Internal(InternalError),
+}
+
+impl DeleteEntityPropertyError {
+    pub fn entity_not_found(entity: &Entity) -> Self {
+        Self::EntityNotFound(entity.into())
+    }
+
+    pub fn property_not_found(entity: &Entity, property_name: PropertyName) -> Self {
+        Self::PropertyNotFound(EntityPropertyNotFoundError {
+            entity_type: entity.entity_type,
+            entity_id: entity.entity_id.to_string(),
+            property_name,
+        })
+    }
+}
+
+#[derive(Error, Debug)]
+#[error("Entity not found: entity_type='{entity_type:?}', entity_id='{entity_id}'")]
+pub struct EntityNotFoundError {
+    pub entity_type: EntityType,
+    pub entity_id: String,
+}
+
+impl From<&Entity<'_>> for EntityNotFoundError {
+    fn from(entity: &Entity<'_>) -> Self {
+        Self {
+            entity_type: entity.entity_type,
+            entity_id: entity.entity_id.to_string(),
+        }
+    }
+}
+
+#[derive(Error, Debug)]
+#[error(
+    "Entity property not found: entity_type='{entity_type:?}', entity_id='{entity_id}', \
+     property_name='{property_name:?}'"
+)]
+pub struct EntityPropertyNotFoundError {
+    pub entity_type: EntityType,
+    pub entity_id: String,
+    pub property_name: PropertyName,
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Error, Debug)]
+pub enum GetEntityPropertiesError {
+    #[error(transparent)]
+    NotFound(EntityNotFoundError),
+
+    #[error(transparent)]
+    Internal(InternalError),
+}
+
+impl GetEntityPropertiesError {
+    pub fn entity_not_found(entity: &Entity<'_>) -> Self {
+        Self::NotFound(entity.into())
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Error, Debug)]
+pub enum InsertEntitiesRelationError {
+    #[error(transparent)]
+    Duplicate(InsertEntitiesRelationDuplicateError),
```

Comment:
Again, isn't this against idempotence?

Reply:
Replied in another thread: https://github.com/kamu-data/kamu-cli/pull/750#discussion_r1713847812